### ISSUE TITLE
Same distribution, nonzero loss? 

### ISSUE THREAD

**Issue by kyuheejo** [facebookresearch/dino#167]
Hi, I observed that my model is failing to converge. I am trying to debug the code and I am observing this peculiar behavior: 

`torch.sum(-F.softmax(student_out[0][0])*F.log_softmax(student_out[0][0], -1), -1)`

returns 6.9058

Shouldn't this theoretically return 0 since both are the same distribution?

**Comment by woctezuma** [comment]
I think it is the entropy forumula here, not the Kullback-Leibler divergence.

**Comment by woctezuma** [comment]
I am just talking about the formula which appears in the first post. It looks like this forumla:

https://en.wikipedia.org/wiki/Entropy_(information_theory)

So I don't expect the result to be zero.

**Comment by kyuheejo** [comment]
I see what you mean, it is indeed cross entropy loss which is the loss for DINO pertaining. What I don't understand is how come the cross entropy loss is training target if our goal is to match representations.

**Comment by woctezuma** [comment]
I see. I have found the piece of code from which the line in the first post was:

https://github.com/facebookresearch/dino/blob/cb711401860da580817918b9167ed73e3eef3dcf/main_dino.py#L380-L404

**Comment by swarajnanda2021** [comment]
A bit late, but were you able to achieve convergence? A lot of issues with the repo seem to stem around this fact. The issue is that even independent reproductions of the method seem to have this issue.
